### PR TITLE
feat(core): 12130 & 12146 - Folding panels with new header design

### PR DESCRIPTION
**Pull request by IlyaIzr**
image
we still use icons for mobile viewports
image

### CODE REVIEW

**Review by Komzpa — APPROVED** [review]
Screenshots LGTM

**Comment by Komzpa** [comment]
```
> disaster-ninja-fe@2.0.35-beta lint:css
> stylelint ./src/**/*.css


src/features/events_list/components/EventsListPanel/EventsListPanel.module.css
 74:1  ✖  Unexpected duplicate selector ".panelBody", first used at line 1  no-duplicate-selectors
```

**Comment by Akiyamka** [comment]
> is that expected?

Nope, i didn't even noticed while working on it :D Will fix it with max height it has before

**Comment by Akiyamka** [comment]
image
position: absolute for controls it always bad idea

File: src/components/Panel/CloseButton/CloseButton.module.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.closeBtn {
+  display: flex;
+  position: absolute;
```
Comment (Akiyamka) [inline src/components/Panel/CloseButton/CloseButton.module.css]: don't do that
image

File: src/components/Panel/Header/Header.module.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.headerTitle {
+  display: flex;
+  align-items: center;
+  gap: var(--unit);
+  & svg {
+    width: 20px;
+    height: 20px;
+  }
```
Comment (Akiyamka) [inline src/components/Panel/Header/Header.module.css]: looks suspicious

File: src/features/advanced_analytics_panel/components/AdvancedAnalyticsPanel/AdvancedAnalyticsPanel.module.css [hunk 3]
```diff
@@ -60,20 +41,47 @@
...
+  .panel.collapse {
+    display: none;
+  }
+
+  .panelIcon {
+    position: absolute;
+    display: flex;
+    z-index: 1;
+
+    /* REMOVE # 11728 */
```
Comment (Akiyamka) [inline src/features/advanced_analytics_panel/components/AdvancedAnalyticsPanel/AdvancedAnalyticsPanel.module.css]: 11728 is done according to fibery 
https://kontur.fibery.io/search/Xya4G#Tasks/Task/Change-'Disasters'-button-style-11728

File: src/features/advanced_analytics_panel/components/AdvancedAnalyticsPanel/AdvancedAnalyticsPanel.module.css [hunk 4]
```diff
@@ -60,20 +41,47 @@
...
+    top: 55px;
+    left: var(--unit);
+
+    &.show {
+      transition: visibility 0s linear 0s, opacity 0.33s linear 0s;
```
Comment (Akiyamka) [inline src/features/advanced_analytics_panel/components/AdvancedAnalyticsPanel/AdvancedAnalyticsPanel.module.css]: as far as I know a Boolean state, it cannot change smoothly

**Comment by IlyaIzr** [comment]
Sorry, i edited comment above ⬆️  while wanted to reply to it. I'll fix analytics height

File: src/components/Panel/Header/Header.module.css (hunk 2 above)
Comment (Akiyamka) [reply src/components/Panel/Header/Header.module.css]: image

File: src/features/advanced_analytics_panel/components/AdvancedAnalyticsPanel/AdvancedAnalyticsPanel.module.css [hunk 5]
```diff
@@ -60,20 +41,47 @@
...
 }
 
 @media screen and (max-width: 960px) {
-  .panel {
+  .panel.show {
     position: absolute;
     margin: 0;
     width: 100vw;
     max-height: unset;
     height: calc(100vh - var(--header-height));
```
Comment (Akiyamka) [inline src/features/advanced_analytics_panel/components/AdvancedAnalyticsPanel/AdvancedAnalyticsPanel.module.css]: bad idea, avoid to use fixed value here, header can be multiline

File: src/features/bivariate_manager/components/BivariatePanel/BivariatePanel.module.css [hunk 6]
```diff
@@ -9,22 +9,21 @@
...
 
   &.show {
-    transition: visibility 0s linear 0s, opacity 0.33s linear 0s,
-      margin-right 0.1s linear 0s;
+    transition: visibility 0s linear 0s, height 0.1s linear 0s;
     visibility: visible;
-    opacity: 1;
-    margin-right: -8px;
+    height: 100%;
+    z-index: 10; /* Show over all other panels */
```
Comment (Akiyamka) [inline src/features/bivariate_manager/components/BivariatePanel/BivariatePanel.module.css]: we have variables for z indexes

File: src/features/bivariate_manager/components/BivariatePanel/BivariatePanel.module.css [hunk 7]
```diff
@@ -9,22 +9,21 @@
...
-  &.hide {
-    transition: visibility 0s linear 0.33s, opacity 0.33s linear 0s,
-      margin-right 0.15s linear 0s;
-    visibility: hidden;
-    opacity: 0;
-    margin-right: calc(-75vw - 8px);
+  &.collapse {
+    transition: visibility 0s linear 0s, height 0.1s linear 0s;
+    visibility: visible;
+    height: var(--panel-header-height);
```
Comment (Akiyamka) [inline src/features/bivariate_manager/components/BivariatePanel/BivariatePanel.module.css:34]: You should collapse card body, now wrapper!

File: src/features/bivariate_manager/components/BivariatePanel/BivariatePanel.tsx [hunk 8]
```diff
@@ -4,25 +4,31 @@ import clsx from 'clsx';
...
+    (intercomApp[0] as HTMLDivElement).style.display = '';
+  }
+}
+function hideIntercom() {
+  const intercomApp = document.getElementsByClassName(INTERCOM_ELEMENT_ID);
+  if (intercomApp && intercomApp.length) {
+    (intercomApp[0] as HTMLDivElement).style.display = 'none';
+  }
+}
+
```
Comment (Akiyamka) [inline src/features/bivariate_manager/components/BivariatePanel/BivariatePanel.tsx]:
```suggestion

const intercomButton = () => {
  const setStyleDisplay = (displayValue) => {
    const intercomApp = document.getElementsByClassName(INTERCOM_ELEMENT_ID);
    if (intercomApp && intercomApp.length) {
      (intercomApp[0] as HTMLDivElement).style.display = displayValue;
    }
  }
  return {
    show: () => setStyleDisplay('block'),
    hide: () => setStyleDisplay('none'),
  }
}
```

File: src/features/bivariate_manager/components/BivariatePanel/BivariatePanel.tsx [hunk 9]
```diff
@@ -32,35 +38,41 @@ export function BivariatePanel({
...
+    hideIntercom();
+  }, [setIsOpen]);
+
+  const togglePanel = useCallback(() => {
+    setIsOpen((wasOpen) => {
+      if (wasOpen) showIntercom();
+      else hideIntercom();
+      return !wasOpen;
+    });
   }, [setIsOpen]);
```
Comment (Akiyamka) [inline src/features/bivariate_manager/components/BivariatePanel/BivariatePanel.tsx]:
```suggestion
  const togglePanel = useCallback(() => {
    setIsOpen((current) => !current);
  }, [setIsOpen]);

  useEffect(() => {
    isOpen ? intercomButton.show() : intercomButton.hide()
  }, [isOpen])
```

File: src/features/legend_panel/components/LegendPanel/LegendPanel.tsx [hunk 10]
```diff
@@ -17,13 +19,14 @@ interface LegendPanelProps {
...
 export function LegendPanel({ layers, iconsContainerRef }: LegendPanelProps) {
   const [isOpen, setIsOpen] = useState<boolean>(false);
 
   const turnOffTooltip = useAction(currentTooltipAtom.turnOffById);
 
-  const onPanelClose = useCallback(() => {
-    setIsOpen(false);
+  const togglePanel = useCallback(() => {
+    setIsOpen((prevState) => !prevState);
     turnOffTooltip(LEGEND_PANEL_FEATURE_ID);
```
Comment (Akiyamka) [inline src/features/legend_panel/components/LegendPanel/LegendPanel.tsx]: it looks like the tooltips should be toggled as well

File: src/features/legend_panel/components/LegendPanel/LegendPanel.tsx [hunk 11]
```diff
@@ -33,14 +36,13 @@ export function LegendPanel({ layers, iconsContainerRef }: LegendPanelProps) {
 
   return (
     <>
-      <PanelWrap onPanelClose={onPanelClose} isPanelOpen={isOpen}>
+      <PanelWrap onPanelClose={() => setIsOpen(false)} isPanelOpen={isOpen}>
```
Comment (Akiyamka) [inline src/features/legend_panel/components/LegendPanel/LegendPanel.tsx]: useCallback

File: src/features/legend_panel/components/LegendPanel/LegendPanel.module.css [hunk 12]
```diff
@@ -93,15 +77,39 @@
...
+
+    &.show {
+      transition: display 0s linear 0s, visibility 0s linear 0s, opacity 0.33s linear 0s;
+      visibility: visible;
+      display: flex;
+      opacity: 1;
+    }
+
+    &.hide {
+      transition: display 0s linear 0.33s, visibility 0s linear 0.33s,
```
Comment (Akiyamka) [inline src/features/legend_panel/components/LegendPanel/LegendPanel.module.css]: extra transition properties

File: src/features/legend_panel/components/LegendPanel/LegendPanel.module.css [hunk 13]
```diff
@@ -93,15 +77,39 @@
...
+    display: none;
   }
-  .legendPanel > div {
-    width: 100%;
+
+  .panelIcon {
+    display: flex;
+
+    &.show {
+      transition: display 0s linear 0s, visibility 0s linear 0s, opacity 0.33s linear 0s;
```
Comment (Akiyamka) [inline src/features/legend_panel/components/LegendPanel/LegendPanel.module.css]: extra transition properties

File: src/features/legend_panel/components/LegendPanel/LegendPanel.tsx [hunk 14]
```diff
@@ -33,14 +36,13 @@ export function LegendPanel({ layers, iconsContainerRef }: LegendPanelProps) {
...
         <Panel
-          header={<Text type="heading-l">{i18n.t('legend')}</Text>}
-          onClose={onPanelClose}
-          className={clsx(s.legendPanel, isOpen && s.show, !isOpen && s.hide)}
-          classes={{
-            header: s.header,
-          }}
+          header={<PanelHeader icon={<Legend24 />} title={i18n.t('legend')} />}
+          onClose={togglePanel}
+          className={clsx(s.legendPanel, isOpen && s.show, !isOpen && s.collapse)}
```
Comment (Akiyamka) [inline src/features/legend_panel/components/LegendPanel/LegendPanel.tsx]:
```suggestion
          className={clsx(s.legendPanel, isOpen ? s.show :  s.collapse )}
```

File: src/features/legend_panel/components/LegendPanel/LegendPanel.module.css [hunk 15]
```diff
@@ -93,15 +77,39 @@
...
+      opacity: 1;
+    }
+
+    &.hide {
+      transition: display 0s linear 0.33s, visibility 0s linear 0.33s,
+        opacity 0.33s linear 0s;
+      visibility: hidden;
+      display: none;
+      opacity: 0;
+      width: 0;
```
Comment (Akiyamka) [inline src/features/legend_panel/components/LegendPanel/LegendPanel.module.css]: extra

File: src/features/advanced_analytics_panel/components/AdvancedAnalyticsPanel/AdvancedAnalyticsPanel.tsx [hunk 16]
```diff
@@ -30,18 +32,18 @@ export function AdvancedAnalyticsPanel() {
 
   return (
     <div className={s.panelContainer}>
-      <PanelWrap onPanelClose={onPanelClose} isPanelOpen={isOpen}>
+      <PanelWrap onPanelClose={() => setIsOpen(false)} isPanelOpen={isOpen}>
```
Comment (Akiyamka) [inline src/features/advanced_analytics_panel/components/AdvancedAnalyticsPanel/AdvancedAnalyticsPanel.tsx]: useCallback

**Comment by Akiyamka** [comment]
image
something bad with legend paddings

File: src/components/Panel/Wrap/PanelWrap.tsx [hunk 17]
```diff
@@ -25,7 +25,7 @@ export function PanelWrap({
   const shouldCollapse = useMediaQuery(COLLAPSE_PANEL_QUERY);
 
   useEffect(() => {
-    if (shouldCollapse && !restrictAutoCollapsing) {
```
Comment (Akiyamka) [inline src/components/Panel/Wrap/PanelWrap.tsx:28]: All of this should be part of panel component 
image

**Review by Akiyamka — CHANGES_REQUESTED** [review]

File: src/components/Panel/Wrap/PanelWrap.tsx (hunk 17 above)
Comment (IlyaIzr) [reply src/components/Panel/Wrap/PanelWrap.tsx:28]: created a task #12405 to do that

File: src/global.css [hunk 18]
```diff
@@ -1,7 +1,13 @@
 @import-normalize;
 
 :root {
-  --header-height: 40px;
+  --app-header-height: 44px;
+  --panel-header-height: 40px;
+  --icon-s: 16px;
+  --icon-m: 20px;
+  --icon-l: 24px;
```
Comment (Akiyamka) [inline src/global.css]: use priginal sizes of icon

**Review by propakov — COMMENTED** [review]
something wrong happens when open bivariate matrix
image

**Comment by propakov** [comment]
another moment I found 
as it was a custom close button on bivariate panel, need to fix header panel position a bit 

._header_pgs27_5 {
    position: absolute;
    right: 10px;
    top: 0px;
    ...
    }

Screenshot 2022-09-16 at 16 20 57

File: src/features/events_list/components/EventsListPanel/EventsListPanel.tsx [hunk 19]
```diff
@@ -113,49 +118,49 @@ export function EventsListPanel({
...
-                    )}
-                    ref={virtuoso}
-                  />
-                ),
-              })}
-            </div>
+      <Panel
+        header={String(i18n.t('disasters'))}
+        headerIcon={<Disasters24 />}
+        className={clsx(s.eventsPanel, isOpen && s.show, !isOpen && s.hide)}
```
Comment (Akiyamka) [inline src/features/events_list/components/EventsListPanel/EventsListPanel.tsx]:
```suggestion
        className={clsx(s.eventsPanel, isOpen ? s.show : s.hide)}
```

File: src/features/layers_panel/components/MapLayersPanel/MapLayersPanel.tsx [hunk 20]
```diff
@@ -18,29 +20,41 @@ export function MapLayerPanel({
...
   const [isOpen, setIsOpen] = useState<boolean>(true);
   const turnOffTooltip = useAction(currentTooltipAtom.turnOffById);
+  const isMobile = useMediaQuery(IS_MOBILE_QUERY);
 
-  const onPanelClose = useCallback(() => {
-    setIsOpen(false);
+  const togglePanel = useCallback(() => {
+    setIsOpen((prevState) => !prevState);
     turnOffTooltip(LAYERS_PANEL_FEATURE_ID);
   }, [setIsOpen]);
```
Comment (Akiyamka) [inline src/features/layers_panel/components/MapLayersPanel/MapLayersPanel.tsx]:
```suggestion
  const togglePanel = useCallback(() => {
    setIsOpen((prevState) => !prevState);
  }, [setIsOpen]);
  
  useEffect(() => {
      !isOpen && turnOffTooltip(LAYERS_PANEL_FEATURE_ID);
  }, [isOpen])
```

File: src/global.css [hunk 21]
```diff
@@ -1,7 +1,12 @@
 @import-normalize;
 
 :root {
-  --header-height: 40px;
+  --app-header-height: 44px;
+  --panel-header-height: 40px;
+  --icon-s: 16px;
+  --icon-l: 24px;
```
Comment (Akiyamka) [inline src/global.css]:
```suggestion
```

File: src/global.css [hunk 22]
```diff
@@ -1,7 +1,12 @@
...
+
+  --hishest-panel-z-index: 10;
```
Comment (Akiyamka) [inline src/global.css]: Move it to theme variables

**Review by Akiyamka — APPROVED** [review]

**Review by propakov — APPROVED** [review]

**Pull request merged**